Features/FindDataById/testPages/data/find_data_page.py

```diff
@@ -0,0 +1,173 @@
+            else:
+                self.send_keys((By.XPATH, self.find_id.format('Case')), str(UserData.user_details['prod'][0]))
+            self.js_click((By.XPATH, self.find_button.format(value_type)))
+        elif value_type == "form":
+            self.click((By.XPATH, self.find_id.format('Form Submission')))
+            url = self.get_current_url()
+            if "staging" in url:
+                self.send_keys((By.XPATH, self.find_id.format('Form Submission')), str(UserData.user_details['staging'][0]))
+            else:
+                self.send_keys((By.XPATH, self.find_id.format('Form Submission')), str(UserData.user_details['prod'][0]))
+            self.js_click((By.XPATH, self.find_button.format(value_type)))
+        self.js_click(self.view_button)
+        url = self.get_current_url()
+        if "staging" in url:
+            self.page_source_contains(UserData.user_details['staging'][0])
+            assert True,("Correct web user page is opened in staging environment for" ,value_type ,"field search ")
+        else:
+            self.page_source_contains(UserData.user_details['prod'][0])
+            assert True,("Correct web user page is opened in production environment for" ,value_type ,"field search ")
+        self.driver.back()
+
+    def validate_group_ids(self, value_type):
+        if value_type == "case":
+            self.click((By.XPATH,self.find_id.format('Case')))
+            url = self.get_current_url()
+            if "staging" in url:
+                self.send_keys((By.XPATH, self.find_id.format('Case')), str(UserData.user_details['staging'][2]))
+            else:
+                self.send_keys((By.XPATH, self.find_id.format('Case')), str(UserData.user_details['prod'][2]))
+            self.js_click((By.XPATH, self.find_button.format(value_type)))
+        elif value_type == "form":
+            self.click((By.XPATH,self.find_id.format('Form Submission')))
+            url = self.get_current_url()
+            if "staging" in url:
+                self.send_keys((By.XPATH, self.find_id.format('Form Submission')), str(UserData.user_details['staging'][2]))
+            else:
+                self.send_keys((By.XPATH, self.find_id.format('Form Submission')), str(UserData.user_details['prod'][2]))
+            self.js_click((By.XPATH, self.find_button.format(value_type)))
+        self.js_click(self.view_button)
+        url = self.get_current_url()
+        if "staging" in url:
+            self.page_source_contains(UserData.user_details['staging'][2])
+            assert True,("Correct group page is opened in staging environment for" ,value_type ,"field search ")
+        else:
+            self.page_source_contains(UserData.user_details['prod'][2])
+            assert True,("Correct group page is opened in production environment for" ,value_type ,"field search ")
+
+
+    def validate_case_form_input_id(self, value_type,value):
```

Comment:
Again, switch tabs before assertion, close it and switch back to current one. optimize the code

Reply:
updated the code.